pkg/status/status.go

```diff
@@ -345,142 +325,9 @@ func getCommonStatus() (map[string]interface{}, error) {
-		dogstatsdStatsJSON := []byte(expvar.Get("dogstatsd").String())
-		dogstatsdUdsStatsJSON := []byte(expvar.Get("dogstatsd-uds").String())
-		dogstatsdUDPStatsJSON := []byte(expvar.Get("dogstatsd-udp").String())
-		dogstatsdStats := make(map[string]interface{})
-		json.Unmarshal(dogstatsdStatsJSON, &dogstatsdStats) //nolint:errcheck
-		dogstatsdUdsStats := make(map[string]interface{})
-		json.Unmarshal(dogstatsdUdsStatsJSON, &dogstatsdUdsStats) //nolint:errcheck
-		for name, value := range dogstatsdUdsStats {
-			dogstatsdStats["Uds"+name] = value
-		}
-		dogstatsdUDPStats := make(map[string]interface{})
-		json.Unmarshal(dogstatsdUDPStatsJSON, &dogstatsdUDPStats) //nolint:errcheck
-		for name, value := range dogstatsdUDPStats {
-			dogstatsdStats["Udp"+name] = value
-		}
-		stats["dogstatsdStats"] = dogstatsdStats
-	}
-
-	pyLoaderData := expvar.Get("pyLoader")
-	if pyLoaderData != nil {
-		pyLoaderStatsJSON := []byte(pyLoaderData.String())
-		pyLoaderStats := make(map[string]interface{})
-		json.Unmarshal(pyLoaderStatsJSON, &pyLoaderStats) //nolint:errcheck
-		stats["pyLoaderStats"] = pyLoaderStats
-	} else {
-		stats["pyLoaderStats"] = nil
-	}
-
-	pythonInitData := expvar.Get("pythonInit")
-	if pythonInitData != nil {
-		pythonInitJSON := []byte(pythonInitData.String())
-		pythonInit := make(map[string]interface{})
-		json.Unmarshal(pythonInitJSON, &pythonInit) //nolint:errcheck
-		stats["pythonInit"] = pythonInit
-	} else {
-		stats["pythonInit"] = nil
-	}
-
-	hostnameStatsJSON := []byte(expvar.Get("hostname").String())
-	hostnameStats := make(map[string]interface{})
-	json.Unmarshal(hostnameStatsJSON, &hostnameStats) //nolint:errcheck
-	stats["hostnameStats"] = hostnameStats
-
-	ntpOffset := expvar.Get("ntpOffset")
-	if ntpOffset != nil && ntpOffset.String() != "" {
-		stats["ntpOffset"], err = strconv.ParseFloat(expvar.Get("ntpOffset").String(), 64)
-	}
-
-	inventories := expvar.Get("inventories")
-	var inventoriesStats map[string]interface{}
-	if inventories != nil {
-		inventoriesStatsJSON := []byte(inventories.String())
-		json.Unmarshal(inventoriesStatsJSON, &inventoriesStats) //nolint:errcheck
-	}
-
-	checkMetadata := map[string]map[string]string{}
-	if data, ok := inventoriesStats["check_metadata"]; ok {
-		for _, instances := range data.(map[string]interface{}) {
-			for _, instance := range instances.([]interface{}) {
-				metadata := map[string]string{}
-				checkHash := ""
-				for k, v := range instance.(map[string]interface{}) {
-					if vStr, ok := v.(string); ok {
-						if k == "config.hash" {
-							checkHash = vStr
-						} else if k != "config.provider" {
-							metadata[k] = vStr
-						}
-					}
-				}
-				if checkHash != "" && len(metadata) != 0 {
-					checkMetadata[checkHash] = metadata
-				}
-			}
-		}
-	}
-	stats["inventories"] = checkMetadata
-	if data, ok := inventoriesStats["agent_metadata"]; ok {
-		stats["agent_metadata"] = data
-	} else {
-		stats["agent_metadata"] = map[string]string{}
-	}
```

Comment:
I'm no longer doing this check inside the inventories collector function https://github.com/DataDog/datadog-agent/blob/4eb0f298e369a3f85158e834e144ea3432edbc1b/pkg/metadata/inventories_collector.go#L25

The reason I'm not doing such a check is that it seems that with the work done by @hush-hush on https://github.com/DataDog/datadog-agent/pull/20057 The inventories payload no longer have the `agent_metadata` information